### PR TITLE
Add in basic support for reading lists from parquet

### PR DESCRIPTION
**Pull request by revans2**
https://github.com/NVIDIA/spark-rapids/issues/1126 is a follow on issue to this so that we can read nested timestamps and dates in all cases.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: shims/spark300db/src/main/scala/com/nvidia/spark/rapids/shims/spark300db/Spark300dbShims.scala [hunk 1]
```diff
@@ -18,14 +18,15 @@ package com.nvidia.spark.rapids.shims.spark300db
...
-import com.nvidia.spark.rapids.GpuOverrides.isSupportedType
 import com.nvidia.spark.rapids._
 import com.nvidia.spark.rapids.shims.spark300.Spark300Shims
+
 import org.apache.spark.sql.rapids.shims.spark300db._
 import org.apache.hadoop.fs.Path
 
 import org.apache.spark.rdd.RDD
 import org.apache.spark.SparkEnv
+
```
Comment (tgravescs) [inline shims/spark300db/src/main/scala/com/nvidia/spark/rapids/shims/spark300db/Spark300dbShims.scala:29]: remove extra newline,

File: sql-plugin/src/main/scala/com/nvidia/spark/RebaseHelper.scala [hunk 2]
```diff
@@ -27,6 +27,7 @@ object RebaseHelper extends Arm {
   private[this] def isDateTimeRebaseNeeded(column: ColumnVector,
                                                 startDay: Int,
                                                 startTs: Long): Boolean = {
+    // TODO update this for nested column checks
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/com/nvidia/spark/RebaseHelper.scala:30]: put jira number in comment

**Review by tgravescs — COMMENTED** [review]
just a couple nits

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@tgravescs I have addressed your review comments.

**Review by tgravescs — APPROVED** [review]

**Pull request merged**